helm/charts/hydra/templates/deployment.yaml

```diff
@@ -191,6 +191,10 @@ spec:
             {{- with $migrationExtraEnv }}
               {{- toYaml . | nindent 12 }}
             {{- end }}
+          {{- if .Values.deployment.securityContext }}
```

Comment:
I think this would require a small refactor. Right now we are using `.Values.deployment.securityContext` to set the context for the main application, the init container may have/require different permissions than the main application. I think we would need to split it in the following way:

```yaml
.Values.deployment.securityContext #Pod level context, applies to all containers in the generated pod
.Values.hydra.securityContext #Container level context, applies to the main hydra container
.Values.job.securityContext #Container level context, applies to the init container
```

Reply:
The migration job alread uses the deployments securityContext: https://github.com/ory/k8s/blob/master/helm/charts/hydra/templates/job-migration.yaml#L78

The cronjob has defined its own securityContext: https://github.com/ory/k8s/blob/master/helm/charts/hydra/templates/janitor-cron-job.yaml#L55

If we want to add a podSecurityContext, we likly have to add that for janitor cronjob and migration job too?

So i guess, to not to change current values an introducing a braking change, it would have to look like:

```
.Values.deployment.initContainerSecurityContext
.Values.deployment.podSecurityContext
.Values.cronjob.janitor,podSecurityContext
```

What do you think?

Reply:
I see your point, and I agree we don't want to introduce breaking changes if we **really** don't need to :) 
In your proposal, would we leave `.Values.deployment.securityContext`? And if so, would that be not pod level (the name would suggest that is applied to the whole deployment) 🤔

Reply:
I'v did a first shot on that (also added resources to the initcontainer).

I stumpled upon: https://github.com/ory/k8s/blob/master/helm/charts/hydra/templates/deployment-watcher.yaml#L46

It uses `.Values.securityContext`, which does not exist in the values.yaml, so i've considered it a bug.
I've changed it to `.Values.watcher.securityContext` and added also `.Values.watcher.podSecurityContext`.

Please tell me if this is wanted, as it's changing for users who actually using it this way?


Yes, i would keep `.Values.deployment.securityContext`. I've names all other vars `podSecurityContext` for that reason.

Reply:
Yes, `.Values.securityContext` was the first iteration when the chats only deployed the single deployment, this must have been an oversight and orphaned :( I think your changes make sense here, and they seem consistent with the current approach :)

Reply:
Let me know if there are any more changes needed and of commits should be squashed before merge?